### ISSUE TITLE
versionsByProtocolVersion index (complement versionsByMinecraftVersion index for protocolVersions lookup)

### ISSUE THREAD

**Issue by deathcap** [PrismarineJS/node-minecraft-data#15]
https://github.com/PrismarineJS/node-minecraft-data/commit/f01605f88054105d89a7efb80371c0d0b7aa4c58 added protocolVersions support with the `versions` array and `versionsByMinecraftVersion` index, but looking up the version data by a protocol version would be useful in some situations. Note that since protocol version has two namespaces (usesNetty: true and usesNetty: false), there is protocol version overlap, but this could be solved by using negative integers for usesNetty: false and usesNetty: true.

Proposed use case: related ot https://github.com/PrismarineJS/node-minecraft-protocol/pull/330 Intra-release multi-protocol support. In node-minecraft-protocol, could allow the `version` options to be either:

* typeof version === 'integer': use a version by protocol version integer (versionsByProtocolVersion index)
* typeof version === 'string': use a version by minecraft release version string (versionsByMinecraftVersion index)

Both indexes would allow looking up majorVersion to map to the minecraft-data 'dataset' (1.8, 1.9...). The Minecraft release version string is more human-readable, suitable for custom client/app development, and the protocol version integer is more programmatically useful for automatically passing from the ping packet for https://github.com/PrismarineJS/node-minecraft-protocol/pull/331 dynamic cross-protocol support.

This index could be used in https://github.com/PrismarineJS/node-minecraft-data/issues/13 Update to use minecraft-data for toMajor(), so these would be equivalent:

```javascript
   require('minecraft-data')('1.8.9')
   require('minecraft-data')(47)
```

(but accessing these indexes may require https://github.com/PrismarineJS/node-minecraft-data/issues/14 Lookup protocol data at top-level of module?)

**Comment by rom1504** [comment]
I actually did that before commiting https://github.com/PrismarineJS/node-minecraft-data/commit/f01605f88054105d89a7efb80371c0d0b7aa4c58 but then I noticed that there are multiple versions for a single protocol version (for example 47 has lot of versions), which one should we return ? all of them ?
About pre/post netty, I was thinking about just separating that index in too, a bit like what's been done for mobs and objects.

**Comment by rom1504** [comment]
I'm going to return all corresponding versions. For toMajor, I guess I'll just pick the first one. Currently the mapping protocolVersion -> majorVersion is unique anyway.

**Comment by rom1504** [comment]
Released as 0.19.0
